example/spec/parallel/ingestGranule/IngestGranuleDuplicateHandlingSpec.js

```diff
@@ -0,0 +1,247 @@
+          const renamedFiles = currentFiles.filter((f) => path.basename(parseS3Uri(f.filename).Key).startsWith(`${fileUpdated}.v`));
+          expect(renamedFiles.length).toEqual(1);
+
+          const expectedRenamedFileSize = existingFiles.filter((f) => f.filename.endsWith(fileUpdated))[0].fileSize;
+          expect(renamedFiles[0].fileSize).toEqual(expectedRenamedFileSize);
+        });
+
+        // There is a delay between workflow completion and the granule appears in dynamodb (sns->dbindexer),
+        // so this check does not always work.
+        // it('captures both files', async () => {
+        //   const granule = await apiTestUtils.getGranule({
+        //     prefix: config.stackName,
+        //     granuleId: inputPayload.granules[0].granuleId
+        //   });
+        //   expect(granule.files.length).toEqual(5);
+        // });
+      });
+
+      describe('encounters data with a duplicated filename with duplicate checksum', () => {
+        it('does not create a copy of the file', async () => {
+          expect(currentFiles.filter((f) => f.filename.endsWith(fileNotUpdated)))
+            .toEqual(existingFiles.filter((f) => f.filename.endsWith(fileNotUpdated)));
+        });
+      });
+    });
+
+    describe('and it encounters data with a duplicated filename with different checksum and there is an existing versioned file', () => {
+      let lambdaOutput;
+      let updatedFileName;
+
+      beforeAll(async () => {
+        lambdaOutput = await lambdaStep.getStepOutput(workflowExecution.executionArn, 'MoveGranules');
+
+        // update one of the input files, so that the file has different checksum
+        const content = randomString();
+        const file = inputPayload.granules[0].files[0];
+        updatedFileName = file.name;
+        const updateParams = {
+          Bucket: config.bucket, Key: path.join(file.path, file.name), Body: content
+        };
+
+        await s3().putObject(updateParams).promise();
+        inputPayload.granules[0].files[0].fileSize = content.length;
+
+        workflowExecution = await buildAndExecuteWorkflow(
+          config.stackName, config.bucket, workflowName, collection, provider, inputPayload
+        );
+      });
+
+      it('does not raise a workflow error', () => {
+        expect(workflowExecution.status).toEqual('SUCCEEDED');
+      });
+
+      it('moves the existing data to a file with a suffix to distinguish it from the new file and existing versioned file', async () => {
+        lambdaOutput = await lambdaStep.getStepOutput(workflowExecution.executionArn, 'MoveGranules');
+        const files = lambdaOutput.payload.granules[0].files;
+        expect(files.length).toEqual(6);
+
+        const renamedFiles = files.filter((f) => f.name.startsWith(`${updatedFileName}.v`));
+        expect(renamedFiles.length).toEqual(2);
+      });
+
+      // it('captures all files', async () => {
```

Comment:
Is this intentionally commented out?

Reply:
I don't know.... I need to go figure that out. Thanks for catching that.